app/controllers/declaration/InlandTransportDetailsController.scala

```diff
@@ -61,19 +63,28 @@ class InlandTransportDetailsController @Inject()(
         formWithErrors => Future.successful(BadRequest(inlandTransportDetailsPage(mode, formWithErrors))),
         form => {
           updateCache(form)
-            .map(_ => navigator.continueTo(mode, nextPage(request.declarationType)))
+            .map(maybeCachedDec => navigator.continueTo(mode, nextPage(maybeCachedDec)))
         }
       )
   }
 
-  private def nextPage(declarationType: DeclarationType): Mode => Call =
-    declarationType match {
-      case DeclarationType.STANDARD | DeclarationType.SUPPLEMENTARY =>
-        controllers.declaration.routes.DepartureTransportController.displayPage
-      case DeclarationType.SIMPLIFIED | DeclarationType.OCCASIONAL =>
-        controllers.declaration.routes.BorderTransportController.displayPage
-    }
+  private def nextPage(maybeCachedDec: Option[ExportsDeclaration]): Mode => Call =
+    getSkipOtherTransportPagesValue(maybeCachedDec)
+      .map(_ => controllers.declaration.routes.ExpressConsignmentController.displayPage _)
+      .getOrElse(controllers.declaration.routes.DepartureTransportController.displayPage)
 
   private def updateCache(formData: InlandModeOfTransportCode)(implicit request: JourneyRequest[AnyContent]): Future[Option[ExportsDeclaration]] =
     updateExportsDeclarationSyncDirect(model => model.copy(locations = model.locations.copy(inlandModeOfTransportCode = Some(formData))))
 }
+
+object InlandTransportDetailsController {
+  val invalidOtherTransportPagesValues = List(FixedTransportInstallations, PostalConsignment)
+
+  def getSkipOtherTransportPagesValue(maybeCachedDec: Option[ExportsDeclaration]): Option[ModeOfTransportCode] =
+    for {
+      dec <- maybeCachedDec
+      transport <- dec.locations.inlandModeOfTransportCode
+      inlandModeOfTransportCode <- transport.inlandModeOfTransportCode
```

Comment:
Wondering if it would be worth to a have a `def` in `ExportsDeclaration` to retrieve `inlandModeOfTransportCode` from `locations`